protocols/kad/src/handler.rs

```diff
@@ -618,12 +612,7 @@ where
     type InboundOpenInfo = ();
 
     fn listen_protocol(&self) -> SubstreamProtocol<Self::InboundProtocol, Self::InboundOpenInfo> {
-        if self.config.allow_listening {
-            SubstreamProtocol::new(self.config.protocol_config.clone(), ())
-                .map_upgrade(Either::Left)
-        } else {
-            SubstreamProtocol::new(Either::Right(upgrade::DeniedUpgrade), ())
-        }
+        SubstreamProtocol::new(self.config.protocol_config.clone(), ()).map_upgrade(Either::Left)
```

Comment:
You should be able to now simplify `InboundUpgrade` to no longer use `Either`.

Reply:
Does it mean I just have to remove ```.map_upgrade(Either::Left)``` from here?

Reply:
Yes and change the `InboundUpgrade` associated type of `KademliaHandler`.